cl_billing_pricing.module

```diff
@@ -639,71 +708,74 @@ function cl_billing_pricing_process_line_item($line_item) {
+          if (cl_billing_pricing_check($action, $li_wrapper)) {
+            $selected_actions[] = $action;
+          }
+        }
       }
     }
   }
 
   // Do the same thing for the order owner, if any.
-  $account_actions = array();
   if ($order) {
-    list(,,$account_type) = entity_extract_ids('user', $li_wrapper->order->owner());
+    list(, , $account_type) = entity_extract_ids('user', $li_wrapper->order->owner);
     $account_instances = field_info_instances('user', $account_type);
     foreach ($account_instances as $fieldname => $instance) {
       if (!empty($pricing_fields[$fieldname])) {
         $new_actions = $li_wrapper->order->owner->{$fieldname}->value();
-        array_walk($new_actions, function($action) use($instance) { $action['component'] = $instance['settings']['component']; });
-        $account_actions = array_merge($account_actions, $new_actions);
+        foreach ($new_actions as $key => $action) {
+          $action['component'] = $instance['settings']['component'];
+          $action['group'] = 'account';
+          // Check the validity of each desired pricing action item.
+          // Actions here are just field items from the cl_billing_pricing field type.
+          if (cl_billing_pricing_check($action, $li_wrapper)) {
+            $selected_actions[] = $action;
+          }
+        }
       }
     }
   }
-
-  // Check the validity of each desired pricing action item.
-  // Actions here are just field items from the cl_billing_pricing field type.
-  $selected_items = array();
-  foreach ($account_items as $action) {
-    $action['group'] = 'account';
-    if (cl_billing_pricing_check($action, $li_wrapper)) {
-      $selected_actions[] = $action;
-    }
-  }
-  foreach ($license_items as $action) {
-    $action['group'] = 'license';
-    if (cl_billing_pricing_check($action, $li_wrapper)) {
-      $selected_actions[] = $action;
-    }
-  }
-
-  // Sort the items by their weight. This is a universal sort for all items, to
-  // provide maximum flexibility.
+  // Sort selected actions by weight.
   usort($selected_actions, function($a, $b) { return $a['weight'] - $b['weight']; });
-
-  foreach ($selection_actions as $action) {
+  // Apply actions to line items.
+  foreach ($selected_actions as $action) {
     cl_billing_pricing_apply($action, $li_wrapper);
   }
+  
+  return;
 }
 
 /**
  * Check if a particular pricing action can apply to a line item based on its
  * condition column.
  *
- * @param $action A pricing action field item.
- *
- * @param $li_wrapper A wrapped line item.
- *
- * @return bool Whether this action item should be applied.
+ * @param $action array
+ *   A pricing action field item.
+ * @param $li_wrapper EntityDrupalWrapper
+ *   A wrapped line item.
+ * @return bool
+ *   Whether this action item should be applied.
  */
 function cl_billing_pricing_check($action, $li_wrapper) {
+  // Skip validity check if the action expired.
+  if ($action['expires'] + 86399 < time()) {
```

Comment:
$action['expires'] is timestamp of beginning of the day (Nov-18, 2016, at 00:00:00). If one sets Nov-18, 2016 as an expiry date, the pricing logic should be valid until Nov-18, 2016, at 23:59:59). Right?